openeo_driver/datacube.py

```diff
@@ -770,9 +771,45 @@ def apply_dimension(
                 or (dimension in {self.DIM_BANDS, self.DIM_PROPERTIES}.intersection(self.get_dimension_names()))
                 and target_dimension is None
             ):
+
+                def check_udf_result(result: openeo.udf.UdfData, require_datacube=False):
+                    if not isinstance(result, openeo.udf.UdfData):
+                        raise ValueError(f"UDF should return UdfData, but got {type(result)}")
+                    feature_list = result.get_feature_collection_list()
+                    if not (feature_list and len(feature_list) == 1):
+                        raise ValueError(
+                            f"UDF should return single feature collection but got {feature_list and len(feature_list)}"
+                        )
+                    if require_datacube:
+                        datacube_list = result.get_datacube_list()
+                        if not (datacube_list and len(datacube_list) == 1):
+                            raise ValueError(
+                                f"UDF should return single datacube but got {datacube_list and len(datacube_list)}"
+                            )
+                        return feature_list[0].data, datacube_list[0].array
+                    return feature_list[0].data, None
+
                 log.warning(
                     f"Using experimental feature: DriverVectorCube.apply_dimension along dim {dimension} and empty cube"
                 )
+                regex = re.compile(r"^def apply_vectorcube\(")
+                if any(regex.match(line) for line in single_run_udf.udf.splitlines()):
```

Comment:
you don't have to split the UDF in lines, you can just do the following I think
```suggestion
                if regex.match(single_run_udf, flags=re.MULTILINE):
```

Reply:
It doesn't look like flags is a supported argument for regex.match.

Reply:
ah yeah, that's an annoying thing with `re` in python, it does not support keyword args, you have to do it positional: 
```suggestion
                if regex.match(single_run_udf, re.MULTILINE):
```

Reply:
I tried all variations for regex.match but it seems like it's not supported:
```
regex = re.compile(r"^def apply_vectorcube\(")
found = regex.match(single_run_udf, re.MULTILINE)
```
```
regex = re.compile(r"^def apply_vectorcube\(", re.MULTILINE)
found = regex.match(single_run_udf)
```

It does work with re.findall however:
`re.findall(r"^def apply_vectorcube\(", single_run_udf, re.MULTILINE)`

Reply:
ok sorry I should not have shared code I didn't try

this works for me:
```python
import re
regex = re.compile("^def foo", re.MULTILINE)

print(regex.search("""
#bla
# def foo
"""))
print(regex.search("""
bla
def foo
"""))
```